crates/primitives/src/col_list.rs

```diff
@@ -319,6 +349,17 @@ impl ColListInline {
     fn undo_mark(&self) -> u64 {
         self.0 >> 1
     }
+
+    /// Returns an equivalent list in heap form instead of inline, and adds `col` to it.
+    /// The capacity of the vec will be `2 * (self.len() + 1)`
```

Comment:
We could choose to round this to a power of two, but this also seems like a reasonable choice.